sympy/assumptions/ask.py

```diff
@@ -9,66 +9,1114 @@
+          number may or may not be real.
+
+        - The "non" facts (i.e., ``Q.nonnegative``, ``Q.nonzero``, and
+          ``Q.nonpositive``) are not equivalent to not the fact, but rather, not
+          the fact *and* ``Q.real``.  For example, ``Q.nonnegative`` means
+          ``~Q.negative & Q.real``. So for example, ``I`` is not nonnegative,
+          nonzero, or nonpositive.
+
+        Examples
+        ========
+
+        >>> from sympy import Q, ask, symbols
+        >>> x = symbols('x')
+        >>> ask(Q.real(x), Q.positive(x))
+        True
+        >>> ask(Q.real(0))
+        True
+
+        References
+        ==========
+
+        .. [1] https://en.wikipedia.org/wiki/Real_number
+
+        """
+        return Predicate('real')
+
+    @memoize_property(predicate_storage)
+    def extended_real(self):
+        r"""
+        Extended real predicate.
+
+        ``Q.extended_real(x)`` is true iff ``x`` is a real number including
+        :math:`{-\infty, \infty}`.
+
+        See documentation of ``Q.real`` for more information about related facts.
+
+        Examples
+        ========
+
+        >>> from sympy import ask, Q, oo, I
+        >>> ask(Q.extended_real(1))
+        True
+        >>> ask(Q.extended_real(I))
+        False
+        >>> ask(Q.extended_real(oo))
+        True
+
+        """
+        return Predicate('extended_real')
+
+    @memoize_property(predicate_storage)
+    def imaginary(self):
+        """
+        Imaginary number predicate.
+
+        ``Q.imaginary(x)`` is true iff ``x`` can be written as a real
+        number multiplied by the imaginary unit ``I``. Please note that ``0``
+        is not considered to be an imaginary number.
+
+        Examples
+        ========
+
+        >>> from sympy import Q, ask, I
+        >>> ask(Q.imaginary(3*I))
+        True
+        >>> ask(Q.imaginary(2 + 3*I))
+        False
+        >>> ask(Q.imaginary(0))
+        False
+
+        References
+        ==========
+
+        .. [1] https://en.wikipedia.org/wiki/Imaginary_number
+
+        """
+        return Predicate('imaginary')
+
+    @memoize_property(predicate_storage)
+    def complex(self):
+        """
+        Complex number predicate.
+
+        ``Q.complex(x)`` is true iff ``x`` belongs to the set of complex
+        numbers.
+
+        Examples
+        ========
+
+        >>> from sympy import Q, Symbol, ask, I, oo
+        >>> x = Symbol('x')
+        >>> ask(Q.complex(0))
+        True
+        >>> ask(Q.complex(2 + 3*I))
+        True
+        >>> ask(Q.complex(oo))
+        False
+
+        References
+        ==========
+
+        .. [1] https://en.wikipedia.org/wiki/Complex_number
+
+        """
+        return Predicate('complex')
+
+    @memoize_property(predicate_storage)
+    def algebraic(self):
+        """
+        Algebraic number predicate.
+
+        ``Q.algebraic(x)`` is true iff ``x`` belongs to the set of
+        algebraic numbers.
```

Comment:
A more precise definition is needed here. "Algebraic" in general depends on the base field, but here, `algebraic` means algebraic over the rationals. More precisely, `a` is algebraic if there is some polynomial in `p(x)\in \mathbb{Q}[x]` such that `p(a) = 0`.